### PR TITLE
Rename various version fields.

### PR DESCRIPTION
**Pull request by davidben**
From the past three times we've bumped the version, reality has
unequivocally told us that the TLS version negotiation, as described in
previous specifications, has been a collossal failure.

The working group decided not to replace the mechanism with a more
sustainable one, so do the next best thing in trying to save TLS from
itself. If the fields are renamed to what they actually are
('server_version' is the true version and 'client_version' is a
maximum), then perhaps we'll have some hope of future implementors
getting it right, if we ever clear through the prior iterations of the
bug.

While I'm here, remove the mention of 'The protocol version the current
record is compatible with' for record_version. record_version is purely
vestigial.

### CODE REVIEW

**Comment by kaduk** [comment]
+1

**Comment by davegarrett** [comment]
This sounds like a good idea, however for the server version I'd go with "negotiated_version" to be more precise. For "record_version", we could also extend the "legacy" prefix convention we've started using and change it to "legacy_record_version" to remind that it's a deprecated field.

**Comment by davidben** [comment]
> negotiated_version

Hrm. We don't name other fields like ServerHello.negotiated_cipher_suite and such. Though if we wanted to be consistent with those fields, it should be ClientHello.max_version rather than ClientHello.max_supported_version. Dunno, maybe that would be better... max_supported_version is a bit wordy.

(Though I do think breaking pattern on the ClientHello version is justified because that field is already inconsistent with everything else in TLS. The version is the only place where the client doesn't offer a list, so everything else already self-describes as a "pick one of these" type of field.)

> legacy_record_version.

Done.
